relational_operators/HashJoinOperator.hpp

```diff
@@ -141,6 +141,12 @@ class HashJoinOperator : public RelationalOperator {
     return output_relation_.getID();
   }
 
+  void doneFeedingInputBlocks(const relation_id rel_id) override {
```

Comment:
If I understand correctly: this fix ensures that `HashJoinOperator` is available for scheduling unless `BuildHashOperator` is done.

More details:
`Foreman::markOperatorFinished` calls `doneFeedingInputBlocks` for all dependent operators. `HashJoinOperator` depends on two operators: `BuildHashOperator` on the build relation and some operator (called `ProbeOp`) on the probe one. Thus, without this fix, once `ProbeOp` has finished execution before `BuildHashOperator` is done, it also calls `doneFeedingInputBlocks`, but incorrectly triggers the semantics that `HashJoinOperator` has all blobs of hash table on the build relation.